network/stub/hub.go

```diff
@@ -13,6 +14,7 @@ import (
 // It maintains a set of network instances and enables them to directly exchange message
 // over the memory.
 type Hub struct {
+	sync.Mutex
```

Comment:
is buffer threadsafe?  if not that should be guarded as well.

Reply:
Yep, already threadsafe.